From f193d68ce753374de30a1d723b9a63bbc700878f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 18 Jun 2013 08:48:14 -0400
Subject: [PATCH 160/160] journalctl: properly print headers of empty journals

(cherry picked from commit 718fe4b1d50f89cbc11318edb0c2c3de9be2ac34)

Conflicts:
	src/journal/journal-file.c
---
 src/journal/journal-file.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/journal/journal-file.c b/src/journal/journal-file.c
index 40d4883..387086a 100644
--- a/src/journal/journal-file.c
+++ b/src/journal/journal-file.c
@@ -2270,6 +2270,15 @@ fail:
         log_error("File corrupt");
 }
 
+static const char* format_timestamp_safe(char *buf, size_t l, usec_t t) {
+        const char *x;
+
+        x = format_timestamp(buf, l, t);
+        if (x)
+                return x;
+        return " --- ";
+}
+
 void journal_file_print_header(JournalFile *f) {
         char a[33], b[33], c[33], d[33];
         char x[FORMAT_TIMESTAMP_MAX], y[FORMAT_TIMESTAMP_MAX], z[FORMAT_TIMESTAMP_MAX];
@@ -2317,8 +2326,8 @@ void journal_file_print_header(JournalFile *f) {
                yes_no(journal_file_rotate_suggested(f, 0)),
                (unsigned long long) le64toh(f->header->head_entry_seqnum),
                (unsigned long long) le64toh(f->header->tail_entry_seqnum),
-               format_timestamp(x, sizeof(x), le64toh(f->header->head_entry_realtime)),
-               format_timestamp(y, sizeof(y), le64toh(f->header->tail_entry_realtime)),
+               format_timestamp_safe(x, sizeof(x), le64toh(f->header->head_entry_realtime)),
+               format_timestamp_safe(y, sizeof(y), le64toh(f->header->tail_entry_realtime)),
                format_timespan(z, sizeof(z), le64toh(f->header->tail_entry_monotonic), USEC_PER_MSEC),
                (unsigned long long) le64toh(f->header->n_objects),
                (unsigned long long) le64toh(f->header->n_entries));
-- 
1.8.3.1

